hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/ByteBufferPositionedReadable.java

```diff
@@ -55,6 +56,10 @@
    * <p>
    * Implementations should treat 0-length requests as legitimate, and must not
    * signal an error upon their receipt.
+   * <p>
+   * This does not change the current offset of a file, and is thread-safe.
+   *
+   * <i>Warning: Not all filesystems satisfy the thread-safety requirement.</i>
```

Comment:
I don't understand what this means. Does it mean the implementation of this interface must guanrantee thread safety despite the fact that not all file systems provide intrinsic thread-safety?

Reply:
I originally copied this from `PositionedReadable`, but after taking a second look, the only two filesystems that implement this interface are `CryptoInputStream` and `DFSInputStream` and both of their implementations are thread safe. So I removed this.